src/stdlib/File.ts

```diff
@@ -28,6 +27,201 @@ export function getMemfsPath(fileUri: string) {
+        },
+        impl: (interpreter: Interpreter, file: BrsString) => {
+            const volume = getVolumeByPath(interpreter, file.value);
+            if (volume === null) {
+                return BrsBoolean.False;
+            }
+
+            const memfsPath = getMemfsPath(file.value);
+            try {
+                volume.unlinkSync(memfsPath);
+                return BrsBoolean.True;
+            } catch (err) {
+                return BrsBoolean.False;
+            }
+        }
+    }
+);
+
+/** Deletes a directory (if empty), return true if successful */
+export const DeleteDirectory = new Callable(
+    "DeleteDirectory",
+    {
+        signature: {
+            args: [
+                {name: "dir", type: ValueKind.String}
+            ],
+            returns: ValueKind.Boolean
+        },
+        impl: (interpreter: Interpreter, dir: BrsString) => {
+            const volume = getVolumeByPath(interpreter, dir.value);
+            if (volume === null) {
+                return BrsBoolean.False;
+            }
+
+            const memfsPath = getMemfsPath(dir.value);
+            try {
+                volume.rmdirSync(memfsPath);
+                return BrsBoolean.True;
+            } catch (err) {
+                return BrsBoolean.False;
+            }
+        }
+    }
+);
+
+/** Creates a directory, return true if successful */
+export const CreateDirectory = new Callable(
+    "CreateDirectory",
+    {
+        signature: {
+            args: [
+                {name: "dir", type: ValueKind.String}
+            ],
+            returns: ValueKind.Boolean
+        },
+        impl: (interpreter: Interpreter, dir: BrsString) => {
+            const volume = getVolumeByPath(interpreter, dir.value);
+            if (volume === null) {
+                return BrsBoolean.False;
+            }
+
+            const memfsPath = getMemfsPath(dir.value);
+            try {
+                volume.mkdirSync(memfsPath);
+                return BrsBoolean.True;
+            } catch (err) {
+                return BrsBoolean.False;
+            }
+        }
+    }
+);
+
+/** Stubbed function for formatting a drive; always returns false */
+export const FormatDrive = new Callable(
+    "FormatDrive",
+    {
+        signature: {
+            args: [
+                {name: "drive", type: ValueKind.String},
+                {name: "fs_type", type: ValueKind.String}
+            ],
+            returns: ValueKind.Boolean
+        },
+        impl: (interpreter: Interpreter, dir: BrsString) => {
+            console.warn("`FormatDrive` is not implemented in `brs`.");
+            return BrsBoolean.False;
+        }
+    }
+);
+
+/** Returns an array of paths in a directory */
+export const ListDir = new Callable(
+    "ListDir",
+    {
+        signature: {
+            args: [
+                {name: "path", type: ValueKind.String}
+            ],
+            returns: ValueKind.Array // TODO: change this to roList when available
```

Comment:
re: your todo – this'll likely be `ValueKind.Object` for `roList`, which matches the docs pretty nicely IMO